android-core/src/main/java/com/mparticle/internal/MessageManager.java

```diff
@@ -525,7 +526,10 @@ public void startUploadLoop() {
         mUploadHandler.sendMessageDelayed(mUploadHandler.obtainMessage(UploadHandler.UPLOAD_MESSAGES, mConfigManager.getMpid()), Constants.INITIAL_UPLOAD_DELAY);
     }
 
-    public void doUpload() {
+    public void doUpload(MParticle.UploadCallback... callbacks) {
```

Comment:
So is the idea of using varargs parameter so that you can call this without a callback so that we don't break the existing public API? I think this is a bit confusing because it implies you can have multiple callbacks but then only uses the first one...

I think it would be better to leave the original `public void doUpload()` method as-is, then add a second method like so:

```
public void doUpload(MParticle.UploadCallback callback) {
    uploadCallback = callback;
    doUpload();
}
```

That should make it clearer without breaking the existing public API.